nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -1092,13 +1093,44 @@ def dummy():
         if enc_output_attn_mask is None:
             enc_output_attn_mask = enc_mask
 
+        if use_memory:
+            inference_key_first = torch.tensor([[True] for _ in range(enc_output.shape[0])])
+            inference_key_then = torch.tensor([[False] for _ in range(enc_output.shape[0])])
+            inference_mem_size = torch.tensor([[256] for _ in range(enc_output.shape[0])])
+
+            # cache memory index
+            memory_ids = [
+                torch.tensor([i for _ in range(enc_output.shape[0])]).unsqueeze(-1)
+                for i in range(num_tokens_to_generate)
+            ]
+
         for i in range(num_tokens_to_generate):
             # No microbatches in decoding. Just the global batch.
             decoder_seq_length = predicted_tokens_dec.size(1)
             dec_mask = predicted_tokens_dec != tokenizer.pad_id
 
-            batch_for_pipeline = [enc_output, enc_output_attn_mask, predicted_tokens_dec, dec_mask]
-            arg_names = ['enc_output', 'enc_output_attn_mask', 'dec_input_ids', 'dec_attn_mask']
+            if not use_memory:
+                batch_for_pipeline = [enc_output, enc_output_attn_mask, predicted_tokens_dec, dec_mask]
+                arg_names = ['enc_output', 'enc_output_attn_mask', 'dec_input_ids', 'dec_attn_mask']
+            else:
+                batch_for_pipeline = [
+                    enc_output,
+                    enc_output_attn_mask,
+                    predicted_tokens_dec[..., i : i + 1],
+                    dec_mask[..., i : i + 1],
+                    inference_key_first if i == 0 else inference_key_then,
```

Comment:
isn't `set_inference_key_value_memory` just a boolean why use `torch.tensor([[True] for _ in range(enc_output.shape[0])])`?